packages/mattermost-redux/src/store/configureStore.dev.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createStore, applyMiddleware, Store} from 'redux';
+import {composeWithDevTools} from 'redux-devtools-extension';
+
+import serviceReducer from '../reducers';
+
+import deepFreezeAndThrowOnMutation from 'mattermost-redux/utils/deep_freeze';
+
+import {Reducer, Action} from 'mattermost-redux/types/actions';
+
+import {GlobalState} from 'mattermost-redux/types/store';
+
+import reducerRegistry from './reducer_registry';
+
+import initialState from './initial_state';
+import {createReducer} from './helpers';
+import {createMiddleware} from './middleware';
+
+/**
+ * Configures and constructs the redux store. Accepts the following parameters:
+ * preloadedState - Any preloaded state to be applied to the store after it is initially configured.
+ * appReducer - An object containing any app-specific reducer functions that the client needs.
+ * persistConfig - Any additional configuration data to be passed into redux-persist aside from the default values.
+ * getAppReducer - A function that returns the appReducer as defined above. Only used in development to enable hot reloading.
+ * clientOptions - An object containing additional options used when configuring the redux store. The following options are available:
+ *     additionalMiddleware - func | array - Allows for single or multiple additional middleware functions to be passed in from the client side.
+ *     enableThunk - bool - default = true - If true, include the thunk middleware automatically. If false, thunk must be provided as part of additionalMiddleware.
+ */
+export default function configureServiceStore(preloadedState: any, appReducer: any, persistConfig: any, getAppReducer: any, clientOptions: any): Store {
+    const baseState = Object.assign({}, initialState, preloadedState);
+
+    const store = createStore(
+        createDevReducer(baseState, serviceReducer, appReducer) as any,
```

Comment:
I had to change these invocations of `createDevReducer` to return `any` as well. There's definitely a better fix for this, but something is really messed up already with all these `any`s thrown around in this file.